syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.paperatus.matrix.proto";
option java_outer_classname = "LedControlProto";

package paperatus;
import "google/protobuf/empty.proto";
import "google/protobuf/wrappers.proto";

service LedControl {
  rpc Initialize (Setup) returns (google.protobuf.BoolValue) {}
  rpc WriteData (LedData) returns (google.protobuf.Empty) {}
}

message LedData {
  repeated uint32 data = 1;
  uint32 width = 2;
  uint32 height = 3;
}

message Setup {
  LedData ledData = 1;
  Mapping mapping = 2;
}

message Mapping {
  repeated uint32 data = 1;
}

